packages/frontend/src/state/lp/hooks.ts

```diff
@@ -19,6 +19,78 @@ const POOL_FEE = 3000
 const MAX_INT_128 = new BigNumber(2).pow(128).minus(1).toFixed(0)
 const x96 = new BigNumber(2).pow(96)
 
+/*** ACTIONS ***/
+
+// Opening a mint and LP position and depositing
+export const useOpenPositionDeposit = () => {
+  const { squeethPool } = useAtomValue(addressesAtom)
+  const address = useAtomValue(addressAtom)
+  const contract = useAtomValue(controllerHelperHelperContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const squeethPoolContract = useAtomValue(squeethPoolContractAtom)
+  const isWethToken0 = useAtomValue(isWethToken0Atom)
+  const openPositionDeposit = useAppCallback(
+    async (squeethToMint: BigNumber, lowerTickInput: number, upperTickInput: number, vaultId: number, collatRatio: number, slippage: number, withdrawAmount: number, onTxConfirmed?: () => void) => {
```

Comment:
Not a blocker. Is there a doc on how to calculate these inputs? It will be beneficial for folks who build FE. Or else this will be just a wrapper around etherscan

Reply:
These inputs don't need to be calculated - they'll be directly inputted by the user! The most a calculation will be is dividing by 100 to get percentage to decimal format, etc.

The only substantial calculation left is converting between the squeethToMint param and if the user inputs an ETH equivalent to LP, but this will depend on UI decisions, so I can add that function later.

Reply:
I will make a doc for the "default" inputs once the design is finalized.